src/expr/src/relation/mod.rs

```diff
@@ -1739,11 +1740,42 @@ impl MirRelationExpr {
         false
     }
 
+    /// Given the ids and values of a LetRec, it computes the subset of ids that are used across
+    /// iterations.
+    pub fn ids_used_across_iterations(
```

Comment:
nit 1: rename this to `recursive_ids`.
nit 2: make this take `&self` as an argument and act only if the `&self` is a `LetRec` variant. You can remove the empty `BTreeSet<LocalId>` otherwise.

Reply:
> nit 1: rename this to recursive_ids.

Done

> nit 2: make this take &self as an argument and act only if the &self is a LetRec variant.

The problem is that then I would have to call it on `relation` (an argument of `action`), but that is already mutably borrowed by the big `match`. So this is why I used the components of LetRec, which are given to me by that `match`.